### PR TITLE
Tracking Product Change History

### PR DESCRIPTION
**Pull request by clintonb**
All changes to the StockRecord, Product and ProductAttributeValue models are now recorded in the database.

XCOM-401

@jimabramson @rlucioni

### CODE REVIEW

File: ecommerce/extensions/catalogue/models.py [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+    def _history_user(self, value):  # pragma: no cover
+        self.changed_by = value
+
+
+class ProductAttributeValue(AbstractProductAttributeValue):
+    changed_by = models.ForeignKey('user.User', null=True, blank=True)
+    history = HistoricalRecords()
+
+    @property
+    def _history_user(self):  # pragma: no cover
```
Comment (rlucioni) [inline ecommerce/extensions/catalogue/models.py:26]: Why is this property necessary?

Comment (rlucioni) [inline ecommerce/extensions/catalogue/models.py:26]: Never mind, I answered my own question. ([Reference](https://django-simple-history.readthedocs.org/en/latest/advanced.html#recording-which-user-changed-a-model))

**Comment by rlucioni** [comment]
:+1:

**Comment by jimabramson** [comment]
@clintonb the migration is missing for stockrecord.

File: ecommerce/extensions/catalogue/admin.py [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+from simple_history.admin import SimpleHistoryAdmin
 
 
-class ProductAdminExtended(ProductAdmin):
-    list_display = ('get_title', 'upc', 'get_product_class', 'structure',
-                    'attribute_summary', 'date_created', 'course')
+class ProductAdminExtended(SimpleHistoryAdmin):
+    list_display = ('get_title', 'upc', 'get_product_class', 'structure', 'attribute_summary', 'date_created', 'course')
+    prepopulated_fields = {"slug": ("title",)}
+    inlines = [AttributeInline, CategoryInline, ProductRecommendationInline]
```
Comment (jimabramson) [inline ecommerce/extensions/catalogue/admin.py:8]: why is product recommendation in there?

Comment (clintonb) [inline ecommerce/extensions/catalogue/admin.py:8]: Copied from the original admin.

Comment (jimabramson) [inline ecommerce/extensions/catalogue/admin.py:8]: ¯_(ツ)_/¯

Comment (clintonb) [inline ecommerce/extensions/catalogue/admin.py:8]: My only intent is to add history support to the admin, not change/remove any other existing functionality. Although we do not use the ProductRecommendation model currently, I see no need to remove it from the admin UI.

**Comment by clintonb** [comment]
@jimabramson Migration added.

**Comment by jimabramson** [comment]
:+1:

**Pull request merged**